### PR TITLE
[WIP] feat: increase refs functionality

### LINKED ISSUE #3352: Increase functionality of component ref
## Feature Request
Add functions to component ref to increase functionality of it
- A method to get the current playback position (time).
- A method to change the playback rate.
- A method to change the audio pan direction.
- A method to check if the video is a livestream or not.
- A method to control the volume/mute the video.
- A method to control/get the target latency.
- A method to get the live latency.

## Why it is needed
This will increase developer experience

### PR DESCRIPTION
**Pull request by danielmark0116**
## Test plan

closes #3352

### CODE REVIEW

**Comment by freeboub** [comment]
Hello, @danielmark0116 
Thank you for the PR, understood you start implementation for your need (which make sense !)
Btw, can you please split in small Prs, It is really easier to review (I am thinking of isLiveStream value for exemple)
And don't forget the document, but I think you have it in mind ! :)

File: ios/Video/RCTVideo.swift [hunk 1]
```diff
@@ -678,6 +697,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
 
     @objc
     func setRate(_ rate: Float) {
+        _player?.rate = rate
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:700]: normally, this is done in applyModifiers

File: ios/Video/RCTVideo.swift [hunk 2]
```diff
@@ -716,6 +736,17 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
         applyModifiers()
     }
 
+    @objc
+    func setVolume(_ volume: Float, _ forceUnmute: Bool) {
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:740]: Is this new function really necessary ?

File: ios/Video/RCTVideo.swift [hunk 3]
```diff
@@ -1254,6 +1285,8 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
 
             if self._videoLoadStarted {
                 all(RCTVideoUtils.getAudioTrackInfo(self._player), RCTVideoUtils.getTextTrackInfo(self._player)).then { audioTracks, textTracks in
+                    self._livestream = _playerItem.duration.isIndefinite
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:1288]: I don't think this is the good way to check if playback is Live.
You can have streams with 30min of backward buffer, In that case, I think you will have a valid duration.
please see:
https://stackoverflow.com/questions/6666478/is-there-a-way-to-distinguish-between-a-live-stream-and-an-on-demand-file-stream

**Review by freeboub — COMMENTED** [review]
few comments added

**Comment by KrzysztofMoch** [comment]
I will close this one due to conflicts and inactivity - feel free to ping me and I will reopen this

**Pull request closed without merging**